tests/Unit/Providers/ImageManagerProviderTest.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace Xefi\Faker\Images\Tests\Unit\Providers;
+
+use Intervention\Image\Drivers\Gd\Driver as GdDriver;
+use Intervention\Image\Drivers\Imagick\Driver as ImagickDriver;
+use Intervention\Image\ImageManager;
+use ReflectionClass;
+use ReflectionMethod;
+use Xefi\Faker\Images\Exceptions\NoImageDriverException;
+use Xefi\Faker\Images\Providers\ImageManagerProvider;
+use Xefi\Faker\Images\Tests\Unit\TestCase;
+
+final class ImageManagerProviderTest extends TestCase
+{
+    protected static function getMethod(string $methodName): ReflectionMethod
+    {
+        $class = new ReflectionClass(ImageManagerProvider::class);
+
+        return $class->getMethod($methodName);
+    }
+
+    public function testSelectDriver(): void
```

Comment:
as said earlier you should decline this test in multiple test because you are

testSelectedDriverWithNoExtension
testSelectedDriverGd, etc